operator/v1/types_ingress.go

```diff
@@ -233,6 +233,101 @@ type LoadBalancerStrategy struct {
 	// +kubebuilder:validation:Required
 	// +required
 	Scope LoadBalancerScope `json:"scope"`
+
+	// subfields holds desired load balancer information specific to the
+	// underlying infrastructure provider.
+	//
+	// If empty, defaults will be applied. See specific LoadBalancerSubfields
+	// fields for details about their defaults.
+	//
+	// +optional
+	Subfields LoadBalancerSubfields `json:"subfields,omitempty"`
+}
+
+// LoadBalancerSubfields holds desired load balancer information specific
+// to the underlying infrastructure provider.
+// +union
+type LoadBalancerSubfields struct {
+	// type is the underlying infrastructure provider for the load balancer.
+	// Allowed values are "AWS", "Azure", "BareMetal", "GCP", "OpenStack",
+	// and "VSphere".
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	// +required
+	Type LoadBalancerProviderType `json:"type"`
+
+	// aws provides configuration settings that are specific to AWS
+	// load balancers.
+	//
+	// If empty, defaults will be applied. See specific AWSLoadBalancerParameters
```

Comment:
`AWSLoadBalancerParameters` → `aws` or `subfields of this field`.

Reply:
Commit `f3a2a4b` addresses this feedback.